data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/controller/MultipartController.java

```diff
@@ -171,6 +200,30 @@ private FormDataMultiPart createFormDataMultiPart(Object header, Object payload)
 
         return multiPart;
     }
+    
+    /**
+     * Retrieves an identity token for the given message.
+     *
+     * @param header the message.
+     */
+    private DynamicAttributeToken getToken(Message header) {
```

Comment:
Is this a method that may be also needed by the MultipartSender classes? As the sender needs a DAT, too. Maybe this could be in a utils oder `MessageFunctions` class in `ids-core.message`?

Reply:
I think so. I stumbled over other duplications as well, e.g. the `resolveConnectorId` method in both extension classes. Probably moving all these to `ids-core` would be something for another PR, to restrict this one to the endpoint module, but I'll open an issue.